docs/tutorials/zero-to-hero.md

```diff
@@ -308,601 +116,78 @@ export function getResponseByKey(key: string): string {
-
-> In the file `src/main.js`
-> - Update the `view` and `change` method collections with our new contract methods
-
-```js
-window.contract = await near.loadContract(nearConfig.contractName, {
-  // find the line below and change it to match
-  viewMethods: ["getResponse", "getResponseByKey", "getOracleQueryParams"],
-  // find the line below and change it to match
-  changeMethods: ["setResponse", "setResponseByKey", "setOracleQueryParams"],
-  sender: window.accountId
-});
-```
-
-> Also in the same file `src/main.js`
-> - Append the code below to the bottom of the file
+If you have any problems up to this point, please let us know on our [discord channel](http://near.chat).
 
+## Step 3: Wire up the frontend to call the new API
+### Update the frontend code
+Now let's update the UI to use the backend code that we wrote in the previous step.
+> Update the smart contract "plumbing" code: in `src/main.js`, update the contract definition. Replace the values of `viewMethods` and `changeMethods` with our new smart contract methods.
 ```js
-async function saveOracleQueryParams() {
-  let url = document.getElementById('url').value
-  let uid = document.getElementById('uid').value
-  let payload = document.getElementById('payload').value
-  let status = document.getElementById("status")
-  // logging for visibility
-  console.log('sending Params to the blockchain')
-  status.innerText = "sending Params to the blockchain"
-
-  await contract.setOracleQueryParams({ url, uid, payload });
-  // logging for visibility
-  console.log('Params saved to the blockchain')
-  status.innerText = "Params saved to the blockchain"
-  setTimeout(() => status.innerText = "", 1500)
-}
+  // Initializing our contract APIs by contract name and configuration.
+  window.contract = await window.near.loadContract(nearConfig.contractName, {
+    // NOTE: This configuration only needed while NEAR is still in development
+    // View methods are read only. They don't modify the state, but usually return some value.
+    viewMethods: ['getResponse'],
+    // Change methods can modify the state. But you don't receive the returned value when called.
+    changeMethods: ['setResponse'],
+    // Sender is the account ID to initialize transactions.
+    sender: window.accountId,
+  });
 ```
-
-Awesome. We now have a front end which allows us to save API params to the blockchain. Easy-peasy right?
-
-Let's check to make sure everything is working by clicking the button `Save API Params to Blockchain`, and then calling `getOracleQueryParams` on the dev console to make sure our params were saved correctly.
-
-> In NEAR Studio
-> - Click **Run** to redeploy the application
->
-> In the application
-> - click **Save API Params to Blockchain**
->
-> In the developer console
-> - Invoke our new contract method using the following line of code
-
+> in `src/index.html`, add the following element `Price of bitcoin: <div id="response"/>` after the button.
```

Comment:
1. There are two buttons, can we say "after the second button"
2. `<div id="response"/>` needs to be `<div id="response"></div>` otherwise you can no longer log out. (I'll share video internally, as Github won't let me upload it)

Reply:
done

Reply:
done